src/Views/BaseElement.elm

```diff
@@ -0,0 +1,146 @@
+
+        autocompleteState =
+            db.elements
+                |> List.filter (\component -> not (List.member component excluded))
+                |> AutocompleteSelector.init toString
+    in
+    [ span [ class "QuantityInputWrapper" ]
+        [ quantityView
+            { disabled = disableQuantity
+            , quantity = baseElement.quantity
+            , onChange =
+                \maybeQuantity ->
+                    case maybeQuantity of
+                        Just quantity ->
+                            updateEvent { baseElement | quantity = quantity }
+
+                        _ ->
+                            updateEvent baseElement
+            }
+        ]
+    , autocompleteState
+        |> selectElement baseElement.element
+        |> selectorView baseElement.element toString
+    , db.countries
+        |> List.sortBy .name
+        |> List.map
+            (\{ code, name } ->
+                option
+                    [ selected (Maybe.map .code baseElement.country == Just code)
+                    , value <| Country.codeToString code
+                    , disabled disableCountry
+                    ]
+                    [ text name ]
+            )
+        |> (::)
+            (option
+                [ value ""
+                , selected (baseElement.country == Nothing)
+                ]
+                [ text <| "Par défaut (" ++ defaultCountry ++ ")" ]
+            )
+        |> select
+            [ class "form-select form-select CountrySelector"
+            , onInput
+                (\val ->
+                    updateEvent
+                        { baseElement
+                            | country =
+                                if val /= "" then
+                                    Country.codeFromString val
+                                        |> (\countryCode -> Country.findByCode countryCode db.countries)
+                                        |> Result.toMaybe
+
+                                else
+                                    Nothing
+                        }
+                )
+            ]
+    , span [ class "text-end ImpactDisplay fs-7" ]
+        [ impact
+            |> Format.formatImpact selectedImpact
+        ]
+    , deleteItemButton { disabled = List.length excluded == 1 } deleteEvent
+    ]
+
+
+deleteItemButton : { disabled : Bool } -> msg -> Html msg
+deleteItemButton { disabled } event =
+    button
+        [ type_ "button"
+        , class "baseElementDelete d-flex justify-content-center align-items-center btn btn-outline-primary"
+        , title "Supprimer ce composant"
+        , onClick event
+        , Attr.disabled disabled
+        ]
+        [ Icon.trash ]
+
+
+selectorView : element -> (element -> String) -> msg -> Html msg
+selectorView selectedElement toString selectElement =
+    button
+        [ class "form-select ElementSelector text-start"
+        , id <| "selector-" ++ toString selectedElement
```

Comment:
Maybe use the id, I'm slightly uncomfortable with any possible character being used as DOM ids

![image](https://github.com/MTES-MCT/ecobalyse/assets/41547/dd3ad5cc-0203-4e95-8f9d-d003e8c4519d)